### PR TITLE
Adds default apk value for for Hockey action

### LINKED ISSUE #3725: Android Hockey upload
Hockey action currently supports Android APK upload, but lacks support for a default value for the APK path. It would be nice to have support for the apk file path created by Gradle action in a similar way Crashlitics has.
```ruby
lane :alpha do
   gradle(task: "assembleRelease")
   hockey(...)
end
```

### PR DESCRIPTION
**Pull request by modzelewski**
Fixes #3725

### CODE REVIEW

File: fastlane/lib/fastlane/actions/hockey.rb [hunk 1]
```diff
@@ -13,17 +13,20 @@ class HockeyAction < Action
       def self.run(options)
         # Available options: http://support.hockeyapp.net/kb/api/api-versions#upload-version
 
+        @options = options
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/hockey.rb]: Why is this necessary?

File: fastlane/lib/fastlane/actions/hockey.rb [hunk 2]
```diff
@@ -172,6 +183,14 @@ def self.author
       def self.is_supported?(platform)
         [:ios, :mac, :android].include? platform
       end
+
+      def self.build_file
+        [
+          @options[:ipa],
+          @options[:apk]
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/hockey.rb]: What if the user provided both values? Maybe we should use the `conflicting_option`

https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/options.rb#L36-L40

File: fastlane/lib/fastlane/actions/hockey.rb [hunk 3]
```diff
@@ -13,17 +13,20 @@ class HockeyAction < Action
...
-          if options[:ipa].to_s.length == 0
-            UI.user_error!("You have to provide an ipa file")
+
+          if build_file.nil?
+            UI.user_error!("You have to provide a build file")
           end
 
-          dsym_path = options[:ipa].gsub('ipa', 'app.dSYM.zip')
+          dsym_path = options[:ipa].to_s.gsub('ipa', 'app.dSYM.zip')
           if File.exist?(dsym_path)
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/hockey.rb]: This will result in `File.exist?("")` if the user didn't provide an `ipa`. Let's change that to 

```
options[:ipa] && File.exist?(dsym_path)
```

File: fastlane/lib/fastlane/actions/hockey.rb [hunk 4]
```diff
@@ -172,6 +183,14 @@ def self.author
       def self.is_supported?(platform)
         [:ios, :mac, :android].include? platform
       end
+
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/hockey.rb]: Feel free to add your own name to the author, using
```ruby
["KrauseFx", "modzelewski"]
```

**Comment by KrauseFx** [comment]
Hey @modzelewski, thanks for the PR, I added some comments before it's ready to be merged :+1:

**Comment by modzelewski** [comment]
Thanks for your input @KrauseFx. Please have a look at my updates.

**Comment by modzelewski** [comment]
I also think there might be some problem with the CI. Similar issue as #3831
The tests seem to pass fine when run locally.

**Comment by KrauseFx** [comment]
Looks great, thanks for updating the PR. Could you squash the commits and set the commit message to something like `Added default values for apk paths in hockey action` :+1: Let me know once the PR is ready :rocket:

**Comment by modzelewski** [comment]
Thanks @KrauseFx. Rebased, squashed and changed commit message :sparkles:

**Comment by KrauseFx** [comment]
This is perfect, thank you so much @modzelewski :+1:

**Comment by KrauseFx** [comment]
:rocket:
